keps/sig-api-machinery/4977-dedicated-priority-level-for-event-requests/README.md

```diff
@@ -0,0 +1,970 @@
+how much a priority level can consume the concurrency. In addition to that,
+borrowing is also instroduced to better utilize spare concurrency. The limits on
+borrowing are two-sided: a given priority level has a limit on how much it may
+borrow and a limit on how much may be borrowed from it. Worth mentioning, APF
+controller periodically populates current concurrency shares for each priority
+level (i.e. concurrency share that can be borrowed or need to be borrowed) by
+taking past usage into consideration. So in combination of those two aspects, a
+priority level's Nomial Share is only an initial value and the actual share
+might increase or decrease according to actual usage with borrowing. However,
+the owner priority level should have full Nominal Share as its current
+concurrency share if in full utilization.
+
+This KEP proposes the following settings:
+
+| Name  | Proposed       | Proposed | Proposed Borrowing | Proposed Guaranteed |
+:       : Nominal Shares : Lendable : Limit              : Shares              :
+| ----- | -------------: | -------: | -----------------: | ------------------: |
+| event | 5              | 0%       | 100%               | 5                   |
+
+Note: The event priority level is specified with Borrowing Limit to prevent it
+from borrowing all spare shares from other priority levels. We may change it to
+none once we have [weighted borrowing] implemented.
+
+[weighted borrowing]: https://github.com/kubernetes/enhancements/pull/4733
+
+In addition to above, a new priority level and flow schema will be created for
+Events:
+
+```
+apiVersion: flowcontrol.apiserver.k8s.io/v1beta3
+kind: FlowSchema
+metadata:
+  name: events
+spec:
+  distinguisherMethod:
+    type: ByUser
+  matchingPrecedence: 450
+  rules:
+  - nonResourceRules:
+    resourceRules:
+    - apiGroups:
+      - ""
+      - events.k8s.io
+      clusterScope: true
+      namespaces:
+      - '*'
+      resources:
+      - events
+      verbs:
+      - create
+      - update
+      - delete
+    subjects:
+    - group:
+        name: system:authenticated
+      kind: Group
+```
+
+Read Verbs are excluded from this new flow schema because Administrators or
+Monitoring may need to read them for troubleshooting/monitoring while the
+cluster is under extreme stress.
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+#### Story 2
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
```

Comment:
And the comment that I brought up in some different place we were discussing it.

I would like to understand how we're going to safely roll this out, given we will effectively change APF configuration for (almost) all clusters in the world.